numcodecs/pcodec.py

```diff
@@ -44,39 +47,70 @@ class PCodec(Codec):
     def __init__(
         self,
         level: int = 8,
+        mode_spec: Literal["auto", "classic"] = "auto",
```

Comment:
```suggestion
        *
        mode_spec: Literal["auto", "classic"] = "auto",
```
Since you're changing the argument order here anyway, it would be a good oppurtunity to make these argumetns keyword only.

Reply:
Yeah this was my biggest hesitation on backwards compatibility when I looked at this again. Alternatively, I'm open to reverting to the previous kwarg order, but it makes the signature a bit messy. Otherwise I think keyword only is a good addition.

Reply:
I think we should either:
- Break the order, and make everything keyword only
- Keep the order, and make everything *new* keyword only

Happy to be led by you on what's the best option? We're still < verison 1, so can break things between releases.

Reply:
@rabernat any opinion here?

Reply:
I think breaking the order and making everything keyword only is the right way to go.

I'm much less concerned about API stability for a new and evolving codec like PCodec than I am about backwards compatibility of actual data on dask.